python/nano/src/bigdl/nano/deps/neural_compressor/inc_api_2.py

```diff
@@ -0,0 +1,170 @@
+        # pop None values to use default
+        if v is not None:
+            not_none_kwargs[k] = v
+
+    q_model = _quantize(model=model, dataloader=dataloader, metric=metric, **not_none_kwargs)
+
+    if 'pytorch' in not_none_kwargs['framework']:
+        from .pytorch.quantized_model import PytorchQuantizedModel
+        quantized_model = PytorchQuantizedModel(q_model, thread_num)
+
+        from bigdl.nano.pytorch.utils import patch_attrs_from_model_to_object
+        patch_attrs_from_model_to_object(model, quantized_model)
+        return quantized_model
+
+    elif 'tensorflow' in not_none_kwargs['framework']:
+        from .tensorflow.model import KerasQuantizedModel
+        return KerasQuantizedModel(q_model)
+
+    elif 'onnx' in not_none_kwargs['framework']:
+        onnxruntime_session_options = not_none_kwargs.get('onnxruntime_session_options', None)
+        onnx_option = not_none_kwargs.get('onnx_option', None)
+        if onnxruntime_session_options is None:
+            import onnxruntime
+            onnxruntime_session_options = onnxruntime.SessionOptions()
+        if thread_num is not None:
+            onnxruntime_session_options.intra_op_num_threads = thread_num
+            onnxruntime_session_options.inter_op_num_threads = thread_num
+        if onnx_option == 'tensorflow':
+            from bigdl.nano.deps.onnxruntime.tensorflow.tensorflow_onnxruntime_model import \
+                KerasONNXRuntimeModel
+            return KerasONNXRuntimeModel(q_model.model,
+                                         onnxruntime_session_options=onnxruntime_session_options)
+        else:
+            from bigdl.nano.deps.onnxruntime.pytorch.pytorch_onnxruntime_model import \
+                PytorchONNXRuntimeModel
+            quantized_model = PytorchONNXRuntimeModel(q_model.model, None,
+                                                      onnxruntime_session_options)
+
+            from bigdl.nano.pytorch.utils import patch_attrs_from_model_to_object
+            patch_attrs_from_model_to_object(model, quantized_model)
+            return quantized_model
+
+    else:
+        invalidInputError(False,
+                          f"Invalid framework argument: {not_none_kwargs['framework']}")
+
+
+def _quantize(
+    model,
+    metric,
+    dataloader,
+    framework,
+    conf=None,
+    approach='static',
+    tuning_strategy='bayesian',
+    accuracy_criterion={'relative': 0.99, 'higher_is_better': True},
+    timeout=0,
+    max_trials=1,
+    inputs=[],
+    outpus=[],
+    **kwargs
+):
+    from neural_compressor import quantization
+    from neural_compressor.quantization import PostTrainingQuantConfig
+    from neural_compressor.config import AccuracyCriterion, TuningCriterion
+    from neural_compressor.conf.pythonic_config import Config
+    from neural_compressor.conf.config import QuantConf
+    from neural_compressor.data import DataLoader
+
+    if approach == "dynamic":
+        dataloader = None
```

Comment:
In my understanding, we don't support "dynamic"?
So if here should not exist "dynamic" approach?

Reply:
We support 'dynamic', and there is a UT for this: <https://github.com/intel-analytics/BigDL/blob/b6aeea34b7b0141839a48f061bf22dfa0c86a2a8/python/nano/test/inc/pytorch/test_quantize.py#L97>

Reply:
we don't support 'dynamic' for tensorflow